src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,23 +550,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            ' PROTOTYPE(continuation-comments): update once LangVersion 16 is created
+            ' TODO the following line Feature.NonTrailingNamedArguments needs to change to be the correct feature name
+            If CanGet(Here + 1) AndAlso IsWhitespace(Peek(Here)) AndAlso IsSingleQuote(Peek(Here + 1)) Then
+                If CheckFeatureAvailability(Feature.PROTOTYPECommnetsAfterLineContinuation) Then
```

Comment:
You want to use another overload of `CheckFeatureAvailability`. The one that takes a feature and a node, and returns a node (potentially with a diagnostic attached).
Then you want to parse unconditionally with the new logic, but you do: 
```
Dim lineContinuation = MakeLineContinuationTrivia(...)
lineContinuation = CheckFeatureAvailability(Feature.CommentsAfterLineContinuation, lineContinuation)
```

So the syntax tree is the same, regardless of the current language version, but that node in the tree gets a diagnostic if the current language version is too old.
This should be validated in a syntax/parsing test with old vs. new language version.

Reply:
I didn't touch the outer "else" part that is the original code and it will report any errors the same as before. I did this because adding a new diagnostic requires translation and I was trying to minimize changes. There could/should be a diagnostic in the else part of the code below to let people know that the new feature exists but that requires a much bigger change.
```
               If CheckFeatureAvailability(Feature.PROTOTYPECommnetsAfterLineContinuation) Then
                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
                    tList.Add(MakeWhiteSpaceTrivia(GetText(Here)))
                    ScanCommentIfAny(tList)
                    ch = Peek()
                    atNewLine = IsNewLine(ch)
                Else
                    ' We should report the existence of new feature here so users can use latest compiler options
                    Return False
                End If
```

Reply:
>  There could/should be a diagnostic in the else part of the code below ...

The parsing logic should be shared between the two cases (old vs. new language version). The only difference should be that a diagnostic is added. In other words, most of the code should be shared between the two cases.

Reply:
I have just updated the code and you do get an error in the else. The message below might need to be changed or split into 2 one for LineContWithComment and one for NoPrecSpace.

My code only returns a Boolean for both cases and the error 
```
  <data name="ERR_LineContWithCommentOrNoPrecSpace" xml:space="preserve">
    <value>Line continuation character '_' must be preceded by at least one white space and must be the last character on the line.</value>
  </data>
```